### PR TITLE
CORDA-3096-db-error-codes-doc

### PR DESCRIPTION
**Pull request by gwr3com**

### CODE REVIEW

File: docs/source/node-database-migration-logging.rst [hunk 1]
```diff
@@ -75,8 +75,22 @@ An example unsuccessful database initialisation log:
...
+.. _node-database-migration-logging-error-codes:
+
+Error codes
+^^^^^^^^^^^
+
+As mentioned above, an error log entry includes a numeric ``<CODE>`` preceded by the ``error_code=`` label. These error codes serve
+as predefined categories grouping potentially many specific errors.  The following codes are currently in use:
+
+* 1 - errors not belonging to any other category
+* 2 - errors thrown as ``ClassNotFoundException`` by the database library; for example, a missing database driver can trigger this type of error
```
Comment (szymonsztuka) [inline docs/source/node-database-migration-logging.rst]: I would remove or put at the ent mention ``ClassNotFoundException`` to focus on resons of the error e.g.: `DataSource cound not be configured  due  to missing JDBC driver or misconfigured dataSourceClassName. `
Also for any other misonfiguration in node.conf `dataSourceProperties` unrelated to `dataSourceClassName` where a related error would be clasiffied?

File: docs/source/node-database-migration-logging.rst [hunk 2]
```diff
@@ -75,8 +75,22 @@ An example unsuccessful database initialisation log:
...
+* 3 - initialization errors, including connection and authentication issues
+* 4 - internal Corda errors encountered during database migration
```
Comment (szymonsztuka) [inline docs/source/node-database-migration-logging.rst]: Any example on this?

File: docs/source/node-database-migration-logging.rst [hunk 3]
```diff
@@ -75,8 +75,22 @@ An example unsuccessful database initialisation log:
...
+* 5 - database migration errors reported by the liquibase library, including conflicting or duplicate change sets as well as invalid DDL data types
```
Comment (szymonsztuka) [inline docs/source/node-database-migration-logging.rst]: Will a wrong Liquibase XML syntax or non existent tag  be reported as this category as well?

Comment (gwr3com) [reply docs/source/node-database-migration-logging.rst]: yes

**Pull request merged**